source/state_representation/include/state_representation/trajectory/TrajectoryBase.hpp

```diff
@@ -0,0 +1,301 @@
+#pragma once
+
+#include <chrono>
+#include <deque>
+#include <eigen3/Eigen/src/Core/Matrix.h>
+
+#include "state_representation/State.hpp"
+#include "state_representation/exceptions/IncompatibleSizeException.hpp"
+
+namespace state_representation {
+
+/**
+ * @class TrajectoryPoint
+ * @brief Struct to represent the base characteristics of a trajectory point
+ */
+struct TrajectoryPoint {
+  std::string name;
+  Eigen::VectorXd data;
+  std::chrono::nanoseconds duration;
+};
+
+/**
+ * @class TrajectoryBase
+ * @brief Base class that offers common functionalities for trajectory classes
+ */
+template<typename TrajectoryT>
+class TrajectoryBase : public State {
+public:
+  /**
+   * @brief Get list of trajectory point durations
+   * @return the list of trajectory point durations
+   */
+  const std::vector<std::chrono::nanoseconds> get_durations() const;
+
+  /**
+   * @brief Get list of trajectory point times from start
+   * @return the list of trajectory point times from start
+   */
+  const std::vector<std::chrono::nanoseconds> get_times_from_start() const;
+
+  /**
+   * @brief Get number of points in trajectory
+   * @return the number of points in trajectory
+   */
+  unsigned int get_size() const;
+
+  /**
+   * @brief Delete the last point from trajectory
+   */
+  void delete_point();
+
+  /**
+   * @brief Delete the last point from trajectory
+   * @param index the index of the point to delete
+   * @throw std::out_of_range if index is out of range
+   */
+  void delete_point(unsigned int index);
+
+  /**
+   * @brief Reset trajectory
+   */
+  virtual void reset();
+
+protected:
+  /**
+   * @brief Empty constructor
+   */
+  explicit TrajectoryBase();
```

Comment:
```suggestion
  explicit TrajectoryBase() = default;
```
Or do you think the `reset` in line 158 is really necessary?

Reply:
nope, not necessary. Removed